homeassistant/components/proxmoxve/__init__.py

```diff
@@ -86,93 +86,103 @@
+                host,
+                port,
+            )
+            return False
+        except ConnectTimeout:
+            _LOGGER.warning("Connection to host %s timed out during setup", host)
+            return False
 
-            hass.data[PROXMOX_CLIENTS][host] = proxmox_client
+        hass.data[PROXMOX_CLIENTS][host] = proxmox_client
 
     await hass.async_add_executor_job(build_client)
 
     coordinators = hass.data[DOMAIN][COORDINATORS] = {}
 
-    # Create a coordinator for each vm/container
-    for host_config in config[DOMAIN]:
-        host_name = host_config["host"]
-        coordinators[host_name] = {}
+    host_name = entry_data["host"]
+    coordinators[host_name] = {}
 
-        proxmox_client = hass.data[PROXMOX_CLIENTS][host_name]
+    proxmox_client = hass.data[PROXMOX_CLIENTS][host_name]
 
-        # Skip invalid hosts
-        if proxmox_client is None:
-            continue
+    proxmox = proxmox_client.get_api_client()
 
-        proxmox = proxmox_client.get_api_client()
+    for node_config in entry_data["nodes"]:
+        node_name = node_config["name"]
+        node_coordinators = coordinators[host_name][node_name] = {}
 
-        for node_config in host_config["nodes"]:
-            node_name = node_config["node"]
-            node_coordinators = coordinators[host_name][node_name] = {}
-
-            for vm_id in node_config["vms"]:
-                coordinator = create_coordinator_container_vm(
-                    hass, proxmox, host_name, node_name, vm_id, TYPE_VM
-                )
+        for vm_id in node_config["vms"]:
+            coordinator = create_coordinator_container_vm(
+                hass, proxmox, host_name, node_name, vm_id, TYPE_VM
+            )
 
-                # Fetch initial data
-                await coordinator.async_refresh()
+            # Fetch initial data
+            await coordinator.async_refresh()
 
-                node_coordinators[vm_id] = coordinator
+            node_coordinators[vm_id] = coordinator
 
-            for container_id in node_config["containers"]:
-                coordinator = create_coordinator_container_vm(
-                    hass, proxmox, host_name, node_name, container_id, TYPE_CONTAINER
-                )
+        for container_id in node_config["containers"]:
+            coordinator = create_coordinator_container_vm(
+                hass, proxmox, host_name, node_name, container_id, TYPE_CONTAINER
+            )
 
-                # Fetch initial data
-                await coordinator.async_refresh()
+            # Fetch initial data
+            await coordinator.async_refresh()
 
-                node_coordinators[container_id] = coordinator
+            node_coordinators[container_id] = coordinator
 
-    for component in PLATFORMS:
-        await hass.async_create_task(
-            hass.helpers.discovery.async_load_platform(
-                component, DOMAIN, {"config": config}, config
-            )
+    # setup platforms
```

Comment:
Store it here in hass data, using something like this:

```suggestion

    hass.data.setdefault(DOMAIN, {})
    hass.data[DOMAIN] = {
       PROXMOX_CLIENT: proxmox_client,
       COORDINATOORS: coordinators,
    }
 

    # setup platforms
```

For 2 reasons: 

- Keeps above code cleaner
- If something goes wrong above, Home Assistant can retry the entry setup without having "rouge" data from a previous attempt.

Reply:
This I'm not sure. 

How would this handle multiple instances of the domain, every one of each with its proxmox_client and coordinators?

In a later comment you suggested `coordinators = hass.data[DOMAIN][entry.entry_id][COORDINATORS]` but here it seems like there is no  `entry.entry_id` object inside `hass.data[DOMAIN]`